components/state-component.js

```diff
@@ -0,0 +1,89 @@
+// state-component.js
+
+var _ = require('underscore');
+var jsonpointer = require('jsonpointer');
+var promiseComponent = require('./promise-component');
+
+/**
+ * Combines payloads, from multiple ports, arriving in an asynchronous
+ * fashion. Each port may include an indexBy function, property, or path
+ * that will be used to determine a key that must match payloads from other ports.
+ * If no indexBy is provided then the payload will match any payload from other ports.
+ * A provided onchange function is called when payload is received (from all
+ * required port) with payload having the same key from all other inPorts.
+ *
+ * Usage:
+ *   stateComponent({
+ *     inPorts: {name:{indexBy:function(payload)}} of port names to port options || [] of port names,
+ *     onchange: function(inPayloads, outPayload) {
+ *         inPayloads = {name:payload} of port names to last payload,
+ *         outPayload = previous result of onchange
+ *     }
+ *   });
+ */
+module.exports = function(def){
```

Comment:
I'm not sure it would necessarily apply as the definition changes from "node" to "component" as it moves down the stack. I'm also not sure adding the term "node" adds clarity to what it is (as the term is generally overloaded).